bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/handler/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,418 @@
+
+    private boolean isPowerOn = false;
+    private int pollingInterval = DEFAULT_POLLING_INTERVAL_SEC;
+
+    public EpsonProjectorHandler(Thing thing, SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channelId);
+            sendDataToDevice(epsonCommand, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        EpsonProjectorConfiguration config = getConfigAs(EpsonProjectorConfiguration.class);
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_PROJECTOR_SERIAL.equals(thingTypeUID)) {
+            device = Optional.of(new EpsonProjectorDevice(serialPortManager, config));
+        } else if (THING_TYPE_PROJECTOR_TCP.equals(thingTypeUID)) {
+            device = Optional.of(new EpsonProjectorDevice(config));
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+
+        pollingInterval = config.pollingInterval;
+        device.ifPresent(dev -> dev.setScheduler(scheduler));
+        updateStatus(ThingStatus.UNKNOWN);
+        schedulePollingJob();
+    }
+
+    /**
+     * Schedule the polling job
+     */
+    private void schedulePollingJob() {
+        cancelPollingJob();
+
+        pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+            List<Channel> channels = this.thing.getChannels();
+            for (Channel channel : channels) {
+                // only query power & lamp time when projector is off
+                if (isPowerOn || (channel.getUID().getId().equals(CHANNEL_TYPE_POWER)
+                        || channel.getUID().getId().equals(CHANNEL_TYPE_LAMPTIME))) {
+                    updateChannelState(channel);
+                }
+            }
+        }, 0, (pollingInterval > 0) ? pollingInterval : DEFAULT_POLLING_INTERVAL_SEC, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Cancel the polling job
+     */
+    private void cancelPollingJob() {
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        cancelPollingJob();
+        closeConnection();
+        super.dispose();
+    }
+
+    private void updateChannelState(Channel channel) {
+        try {
+            if (!isLinked(channel.getUID()) && !channel.getUID().getId().equals(CHANNEL_TYPE_POWER)) {
```

Comment:
power state will be queried even if the channel isn't linked?

Reply:
Yes for two reasons:
1. If no channels are linked, power is still checked as a ping of the device to set the status to ONLINE and later OFFLINE if the connection is lost.
2. In the edge case of the user not linking the power channel, the power status still needs to be known to properly set the isPowerOn boolean (ie: only PWR? and LAMP? queries can be done when the power is off, because all others would fail).